File: benchmarks/node/03_sieve.mjs
function sieve(n) {
    const flags = new Uint8Array(n + 1).fill(1);
    for (let p = 2; p * p <= n; p++) {
        if (flags[p]) {
            for (let j = p * p; j <= n; j += p) flags[j] = 0;
        }
    }
    let count = 0;
    for (let i = 2; i <= n; i++) if (flags[i]) count++;
    return count;
}
console.log(sieve(10000000));

File: benchmarks/python/03_sieve.py
def sieve(n):
    flags = [True] * (n + 1)
    p = 2
    while p * p <= n:
        if flags[p]:
            j = p * p
            while j <= n:
                flags[j] = False
                j += p
        p += 1
    return sum(1 for i in range(2, n + 1) if flags[i])
print(sieve(10000000))
